lib/components/primitive-components/Group/Group.ts

```diff
@@ -582,18 +584,28 @@ export class Group<Props extends z.ZodType<any, any, any> = typeof groupProps>
     // the Circuit JSON spec is ambiguous as to whether a via should have a
     // separate element from the route)
     for (const pcb_trace of output_pcb_traces) {
-      for (const point of pcb_trace.route) {
-        if (point.route_type === "via") {
-          db.pcb_via.insert({
-            pcb_trace_id: pcb_trace.pcb_trace_id,
-            x: point.x,
-            y: point.y,
-            hole_diameter: 0.3,
-            outer_diameter: 0.6,
-            layers: [point.from_layer as LayerRef, point.to_layer as LayerRef],
-            from_layer: point.from_layer as LayerRef,
-            to_layer: point.to_layer as LayerRef,
-          })
+      if (pcb_trace.type === "pcb_via") {
+        // TODO handling here- may need to handle if redundant with pcb_trace
+        // below (i.e. don't insert via if one already exists at that location)
```

Comment:
CC @imrishabh18 , the output from the server can sometimes have vias that we need to handle, as indicated in the comment we may need some special handling to avoid duplicates. Ideally, we avoid vias defined like this- it's a better better if they're defined as part of the pcb_trace route (that way you could e.g. remove the route with all vias included)

![image](https://github.com/user-attachments/assets/21c43afe-b5fd-462f-9b74-80b022528a9c)

Reply:
Will look more onto this issue, cause now I see the via's are getting skipped.

![image](https://github.com/user-attachments/assets/95b8a821-c2b5-4666-be71-abc6dcce1c81)